src/simulation/LoadGenerator.cpp

```diff
@@ -915,6 +927,46 @@ LoadGenerator::getNextAvailableAccount()
     uint64_t sourceAccountId = *it;
     mAccountsAvailable.erase(it);
     releaseAssert(mAccountsInUse.insert(sourceAccountId).second);
+
+    if (mApp.getHerder().sourceAccountPending(
+            findAccount(sourceAccountId, ledgerNum)->getPublicKey()))
+    {
+        // Although mAccountsAvailable shouldn't contain pending accounts, it is
+        // possible when the network is overloaded. Consider the following
+        // scenario:
+        // 1. This node generates a transaction `t` using account `a` and
+        //    broadcasts it on. In doing so, loadgen marks `a` as in use,
+        //    removing it from `mAccountsAvailable.
+        // 2. For whatever reason, `t` never makes it out of the queue and this
+        //    node bans it.
+        // 3. After some period of time, this node unbans `t` because bans only
+        //    last for so many ledgers.
+        // 4. Loadgen marks `a` available, moving it back into
+        //    `mAccountsAvailable`.
+        // 5. This node hears about `t` again on the network and (as it is no
+        //    longer banned) adds it back to the queue
+        // 6. getNextAvailableAccount draws `a` from `mAccountsAvailable`.
+        //    However, `a` is no longer available as `t` is in the transaction
+        //    queue!
+        //
+        // In this scenario, returning `a` results in an assertion failure
+        // later. At this point we have two reasonable options:
+        // 1. Mark `a` in-use and resample new accounts until we either find one
+        //    that is available, or run out of accounts
+        // 2. Fail loadgen.
+        // We chose option (2), as this scenario is indicative of an overloaded
+        // network and in practice loadgen is highly likely to fail soon anyway.
+        // Moreover, option (1) provides some false sense of having resolved the
+        // issue as this is a race condition that could still occur between
```

Comment:
we're creating and submitting transactions synchronously, so I think there's currently no race condition here, right?

Reply:
Hmm, I was thinking that in between picking `a` and submitting the new transaction, the old transaction `t` could come in on the overlay. But I think you're right this isn't a problem because there's only one thread here and the overlay can't interleave between between these two operations. I'll change this to sample a new account instead.

Reply:
I just pushed a change that resamples instead